### ISSUE TITLE
Regarding validation error message position

### ISSUE THREAD

**Issue by converge** [Rocketseat/unform#106]
**Description of issue**

I'm not able to position the <span>Yup validation error message</span>

**Usage example**
This is an <Input> element:
```js
<Input name="username" type="text" placeholder="Username" />
```

And I'm using this validation for the username field:
```js
const schema = Yup.object().shape({
  username: Yup.string().required('Username is required')
})
```
The validation is called when the `username` field is empty and the message `Username is required` show up just after the `<Input>` field. After the validation message is called this is what I have:
```js
<input type="text" placeholder="Username" id="username" name="username" aria-label="username" value="">
<span>Username is required</span>
```

Is it possible to position the validation message in another place?

**Comment by diego3g** [comment]
Hi @converge, where do you want to show the message? If it's like above the input you can use `order` property from CSS with `display: flex` around the input and the error to achieve that.

Do you have any suggestions about position options we could give to the user?

**Comment by converge** [comment]
I'm used to work with Formik approach, and it's quite nice:
```html
<ErrorMessage name='email' className='field-validation' component='div' />
```
You have this ErrorMessage component, in this way, we're free for positioning the error message/element anywhere, just referencing it by name.

What are your thoughts?

**Comment by diego3g** [comment]
@converge I think it would be good if you could give us a real example showing where you want to put error messages. 

I know Formik is very elastic as it exposes API's for almost everything, but each of these components need to listen to input changes so many renders are executed in background. 

The main focus of unform is to be performatic so we need to study it limitations and see if we can add more API's to handle these edge cases like error message location.

**Comment by joaogn** [comment]
I think I could pass one or more className to span through the parent input. It is possible?I think I could pass one or more className to span through the parent input. It is possible?

**Comment by diego3g** [comment]
@joaogn Today we don't supporting passing any props to error span, but we're welcome to receive contributions.

**Comment by bsmayer** [comment]
I'm missing that as well. Unform is rich on making our lives simple while creating forms, but poor on error customisation. Not all forms can display errors using <span>, and that's when things get complicated. While having error handling automatically done by the library is good in many situations, there are others that are not really positive. 

It might be a good idea to have a way to tell Unform to not display the errors below the fields, but instead pass the list of errors in the submit function, like:

```
<Form 
  enableDefaultValidation={false} 
  onSubmit={(data: any, errors: Array<{ field: string, error: string }>) => {}}>
...
</Form>
```

This way we would be able to grab all the errors and handle them in the way we want, while keeping the current behaviour for the other cases.

**Comment by rserafim** [comment]
Tbm estou com problemas para usá-lo no theme Argon, pois o erro é inserido numa div.

**Comment by KesleyDavid** [comment]
One suggestion would be something to handle javascript errors.

For example:
1 - I click submit
2 - Yup Verification Finds Validation Failure
3 - Instead of appearing a message below the input, have an error "observer".
4 - Send and handle this error as desired by the application

**Comment by diego3g** [comment]
Unform released version 2.0 🎉

Now the library does not includes it's own components of Input, Select, etc.

So you need to create your custom input with Unform's `useField` API.

This way, you can customize the error position and all the other input styles.

Check [this guide](https://unform.dev/recipes/simple-input) to start creating your custom input.
